crmsh/ui_cluster.py

```diff
@@ -283,27 +284,18 @@ def looks_like_hostnames(lst):
                 tie_breaker=options.qdevice_tie_breaker,
                 tls=options.qdevice_tls,
                 cmds=options.qdevice_heuristics)
+        boot_context.qdevice = qdevice
 
-        bootstrap.bootstrap_init(
-            cluster_name=options.name,
-            ui_context=context,
-            nic=options.nic,
-            ocfs2_device=options.ocfs2_device,
-            shared_device=options.shared_device,
-            sbd_device=options.sbd_device,
-            diskless_sbd=options.diskless_sbd,
-            quiet=options.quiet,
-            template=options.template,
-            admin_ip=options.admin_ip,
-            yes_to_all=options.yes_to_all,
-            no_overwrite_sshkey=options.no_overwrite_sshkey,
-            unicast=options.unicast,
-            second_hb=options.second_hb,
-            ipv6=options.ipv6,
-            watchdog=options.watchdog,
-            qdevice=qdevice,
-            stage=stage,
-            args=args)
+        for opt in vars(options):
+            if opt.startswith("qdevice"):
```

Comment:
Do you really need to do this `opt.startswith("qdevice")` check?
Why don't you just update the order of the code, and manage qdevice after this loop?

Reply:
This way you could have the same functionality you have in join and remove

Reply:
No, no matter what's the order of this block of code, since it was a loop of `options`, qdevice option will be included always

Reply:
Solution might be, move this block of `corosync.Qdevice` into function `bootstrap.bootstrap_init`